src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -369,7 +369,17 @@ public static void Sleep(int millisecondsTimeout)
 
 #if !NATIVEAOT
         /// <summary>Returns the operating system identifier for the current thread.</summary>
-        internal static ulong CurrentOSThreadId => GetCurrentOSThreadId();
+        internal static ulong CurrentOSThreadId
+        {
+            get
+            {
```

Comment:
This change does not look right. It should be reverted. Getting current OS thread ID should not depend  in any way on the type of threadpool used.